server/server.go

```diff
@@ -236,8 +239,13 @@ func (s *IDPServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 // newMux creates the HTTP request multiplexer
 // Migrated from legacy/tsidp.go:674-687
-func (s *IDPServer) newMux() *http.ServeMux {
+func (s *IDPServer) newMux() http.Handler {
+
 	mux := http.NewServeMux()
+
+	protect := csrf.New()
+	protect.AddTrustedOrigin(s.serverURL)
```

Comment:
nit: move this to the bottom, right before the `return` where it's actually needed